### PR TITLE
As per the evaluation criteria Completed the home-work-5 

### PR DESCRIPTION
**Pull request by rachapathin**
- Completed all the sub tasks except twitter strategy.
- Added local , Facebook and Google strategies.
- Added JWT creation and validations

### CODE REVIEW

File: controllers/controller.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+const jwt = require("jsonwebtoken");
```
[the remark this answers is not in the document]
Comment (rachapathin) [reply controllers/controller.js]: Resolved

File: controllers/controller.js [hunk 2]
```diff
@@ -18,3 +20,43 @@ export const getAllUsers = (req, res) => {
     res.send("ALL Users"); 
 }
 
+export const checkAuthentication = (req, res) => {
+    const reqBody = req.body;
+    if( reqBody.username === 'nageswar' && reqBody.password === '123456') {
```
[the remark this answers is not in the document]
Comment (rachapathin) [reply controllers/controller.js]: Resolved

File: new-app.js [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
+// Passport strategy block
+app.use(passport.initialize());
+passportStrategy(loginCredentails);
 
 // Routes configuration
-routes(app);
+routes(app, jwtVerify);
 
 // Default Route 
 app.get('/', (req, res) =>
```
[the remark this answers is not in the document]
Comment (rachapathin) [reply new-app.js]: Already created PORT environment variable in index.js

[the remark this answers is not in the document]
Comment (AliakseiBychyk) [reply new-app.js]: You did not created the PORT env variable in 'index.js'
`const port = process.env.PORT || 8080; `
means you are retrieving PORT env variable and assign it to the variable `port`, if there is no PORT env variable `port` is assigned with 8080 value.
So in `new-app.js` you also have to read PORT from `process.env` or if there is no such env variable, to assign the default value.

[the remark this answers is not in the document]
Comment (rachapathin) [reply new-app.js]: Added .env file to maintain all the environment variables.

Resolved

**Pull request merged**